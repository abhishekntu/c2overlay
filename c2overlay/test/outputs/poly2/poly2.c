/****************************************************************/
/*   poly2		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
int foo(int x,int y)
{	
int t1_1,t1_2,t1_3,t1_4,t1_5,t1_6,t1_7,t1_8,t1_9,t1_10,t2_1,t2_2,t2_3,t2_4,t2_5,t2_6,t2_7,t2_8,t3_1,t3_2,t3_3,t3_4,t3_5,t3_6,t3_7,t4_1,t4_2,t4_3,t4_4,t5_1,t5_2,t5_3,t6_1,t6_2,t6_3,t7_1,t7_2,t8_1,t8_2,t9_1,t10_1,t11_1;

t1_1 = y - 1;
t2_1 = x * x;
t2_2 = t1_1 * y;
t3_1 = 3 * y;
t4_1 = 2*t2_1 - t3_1;
t4_2 = t2_2 * t2_2;
t5_1 = t2_1 * t4_1;
return(t5_1 + t4_2);

}
#endif


#ifdef TEST
int main(void)
{
	int x,y;
 	int result;
  	result = foo(x,y);
  	return 0;
}
#endif
