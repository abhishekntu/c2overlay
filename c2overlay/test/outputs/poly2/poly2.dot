digraph graphname {
N8 [color="black", label="N7N3N10_sub_N7_mul_N3_add_N10_N8"];
N9 [color="black", label="O0_N12_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="N11_sqr_N11_N3"];
N4 [color="black", label="N8N5_sub_Imm_1_N8_mul_N5_N4"];
N5 [color="black", label="N9_sqr_N9_N5"];
N6 [color="black", label="N4_mul_Imm_2_N4_N6"];
N7 [color="black", label="N6_mul_Imm_3_N6_N7"];
N8 -> N9;
N1 -> N5;
N2 -> N7;
N2 -> N4;
N3 -> N8;
N4 -> N3;
N5 -> N6;
N5 -> N8;
N6 -> N8;
N7 -> N8;
}
