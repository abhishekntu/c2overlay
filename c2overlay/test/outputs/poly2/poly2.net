.input N1
pinlist: N1
    
.input N2
pinlist: N2
    
.output out:N10
pinlist: N10
    
.clb N8_blk
pinlist: N4 N2 open open N8 open open open open 
subblock: N8_blk 0 1 open open 4 open open open open

.clb N9_blk
pinlist: N2 open open open N9 open open open open 
subblock: N9_blk 0 open open open 4 open open open open

.clb N3_blk
pinlist: N8 open open open N3 open open open open 
subblock: N3_blk 0 open open open 4 open open open open

.clb N4_blk
pinlist: N2 open open open N4 open open open open 
subblock: N4_blk 0 open open open 4 open open open open

.clb N5_blk
pinlist: N1 open open open N5 open open open open 
subblock: N5_blk 0 open open open 4 open open open open

.clb N6_blk
pinlist: N3 N5 N7 open N10 open open open open 
subblock: N6_blk 0 1 2 open 4 open open open open
    
.clb N7_blk
pinlist: N5 N9 open open N7 open open open open 
subblock: N7_blk 0 1 open open 4 open open open open

