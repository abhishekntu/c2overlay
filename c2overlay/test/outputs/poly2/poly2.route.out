Array size: 5 x 5 logic blocks.

Routing:

Net 0 (N1)

SOURCE (5,0)  Pad: 1  
  OPIN (5,0)  Pad: 1  
 CHANX (5,0)  Track: 1  
  IPIN (5,1)  Pin: 2  
  SINK (5,1)  Class: 0  


Net 1 (N2)

SOURCE (1,0)  Pad: 1  
  OPIN (1,0)  Pad: 1  
 CHANX (1,0)  Track: 1  
  IPIN (1,1)  Pin: 2  
  SINK (1,1)  Class: 0  
  OPIN (1,0)  Pad: 1  
 CHANX (1,0)  Track: 0  
 CHANX (2,0)  Track: 0  
 CHANY (2,1)  Track: 0  
 CHANY (2,2)  Track: 0  
  IPIN (3,2)  Pin: 3  
  SINK (3,2)  Class: 0  


Net 2 (N9)

SOURCE (3,1)  Class: 1  
  OPIN (3,1)  Pin: 6  
 CHANX (3,0)  Track: 0  
  IPIN (3,0)  Pad: 0  
  SINK (3,0)  Pad: 0  


Net 3 (N3)

SOURCE (2,1)  Class: 1  
  OPIN (2,1)  Pin: 5  
 CHANY (2,1)  Track: 1  
  IPIN (3,1)  Pin: 3  
  SINK (3,1)  Class: 0  


Net 4 (N4)

SOURCE (1,1)  Class: 1  
  OPIN (1,1)  Pin: 5  
 CHANY (1,1)  Track: 1  
  IPIN (2,1)  Pin: 3  
  SINK (2,1)  Class: 0  


Net 5 (N5)

SOURCE (5,1)  Class: 1  
  OPIN (5,1)  Pin: 7  
 CHANY (4,1)  Track: 1  
  IPIN (4,1)  Pin: 1  
  SINK (4,1)  Class: 0  
 CHANY (4,1)  Track: 1  
 CHANX (4,0)  Track: 1  
 CHANX (3,0)  Track: 1  
  IPIN (3,1)  Pin: 2  
  SINK (3,1)  Class: 0  


Net 6 (N6)

SOURCE (4,1)  Class: 1  
  OPIN (4,1)  Pin: 7  
 CHANY (3,1)  Track: 1  
  IPIN (3,1)  Pin: 1  
  SINK (3,1)  Class: 0  


Net 7 (N7)

SOURCE (3,2)  Class: 1  
  OPIN (3,2)  Pin: 6  
 CHANX (3,1)  Track: 1  
  IPIN (3,1)  Pin: 0  
  SINK (3,1)  Class: 0  
